saltproc/saltproc.py

```diff
@@ -205,53 +422,71 @@ def read_res(self, moment):
+            if true, gets the boc composition
 
         Returns:
         --------
-        bumat_dict: dictionary
-            key: isotope name
-            value: adens
-        mat_def: str
-            material definition in SERPENT with volume and density
+        dep_dict: dictionary
+            key: material name
+            value: dictionary
+                key: isotope
+                value: adens
         """
-        bumat_filename = os.path.join('%s.bumat%i' % (self.input_file, moment))
-        bumat_dict = OrderedDict({})
-        # save isonames for mat file generation
-        self.isoname = []
-
-        with open(bumat_filename, 'r') as data:
-            # this should be changed for two region flows
-            # and in general, hardcoding things is never a good thing
-            for line in itertools.islice(
-                    data, 5, 6):    # Read material description in variable
-                mat_def = line.strip()
-            for line in itertools.islice(
-                    data, 0, None):  # Skip file header start=6, stop=None
-                p = line.split()
-                self.isoname.append(p[0])
-                if '.' in p[0]:
-                    iso = p[0].split('.')[0] + '0'
-                    iso = nucname.name(iso)
-                else:
-                    iso = nucname.name(p[0])
-                if iso[-1] == 'M':
-                    metastable_state = p[0][-1]
-                    # if metastable, label it with state
-                    iso = iso + '-' + str(metastable_state)
-                bumat_dict[iso] = float(p[1])
-        self.isolib_db = bumat_dict.keys()
-        return bumat_dict, mat_def
+        dep_file = os.path.join('%s_dep.m' % self.input_file)
+        with open(dep_file, 'r') as f:
+            lines = f.readlines()
+            self.dep_dict = OrderedDict({})
+            read = False
+            for line in lines:
+                if 'MAT'in line and 'MDENS' in line:
+                    key = line.split('_')[1]
+                    read = True
+                    self.dep_dict[key] = [0] * len(self.isoname)
+                elif read and ';' in line:
+                    read = False
+                elif read:
+                    z = line.split(' ')
+                    # last burnup stage
+                    indx = z.index('%')
+                    mdens = z[indx-1]
+                    if boc:
+                        mdens = z[0]
+                    # the isotope name is at the end of the line.
+                    name = z[-1].replace('\n', '')
+                    # find index so that it doesn't change
+                    try:
+                        where_in_isoname = self.isoname.index(name)
+                        self.dep_dict[key][where_in_isoname] = float(mdens)
+                    except ValueError:
+                        if name not in ['total', 'data']:
+                            print('THIS WAS NOT HERE %s' % name)
```

Comment:
Error messages don't need to be in all caps. 
It's also better if they have information "THIS WAS NOT HERE" is missing a lot of information one typically wants in an error message:

- what is this
- where is here
- why isn't it here?
- what was expected?
- what should the user do next?